### PR TITLE
Tests for DataUtilities' CreateNumberArrays and GetCumulativePercentages

### PR DESCRIPTION
**Pull request by kkcmah**
Check for test overlapping and correctness.

### CODE REVIEW

File: Tests/CreateNumberArray2DTests.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+	}
+	
+	/**
+	 * testing null input array
+	 */
+	@Test(expected = InvalidParameterException.class)
+	public void testNullInputArrayOfArrays() {
+		data = null;
+		createNumberArray2D(data);
+	}
```
Comment (enochtsang) [inline Tests/CreateNumberArray2DTests.java]: What's the expectation here? Every tests should have an expected result.

File: Tests/GetCumulativePercentagesTests.java [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+				allowing(data).getKey(1); will(returnValue(1));
+				allowing(data).getKey(2); will(returnValue(2));
+				allowing(data).getKeys(); will(returnEnumeration(Arrays.asList(0,1,2)));
+				allowing(data).getValue(0); will(returnValue(3));
+				allowing(data).getValue(1); will(returnValue(1));
+				allowing(data).getValue(2); will(returnValue(6));
+				allowing(data).getItemCount(); will(returnValue(3));
+
+			}
+			});
```
Comment (enochtsang) [inline Tests/GetCumulativePercentagesTests.java]: Not a big deal if you're too lazy to fix, but fixing the whitespace here would be nice.

File: Tests/GetCumulativePercentagesTests.java [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+			
+		KeyedValues actualCumulativePercentages = getCumulativePercentages(data);
+		assertEquals("Getting first value result", 0.3, actualCumulativePercentages.getValue(0));
+		assertEquals("Getting second value result", 0.4, actualCumulativePercentages.getValue(1));
```
Comment (enochtsang) [inline Tests/GetCumulativePercentagesTests.java]: Why are these `0.3` and `0.4`, shouldn't it be `1/3`?

File: Tests/GetCumulativePercentagesTests.java [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+				allowing(data).getValue(2); will(returnValue(6));
+				allowing(data).getItemCount(); will(returnValue(3));
+
+			}
+			});
+			
+		KeyedValues actualCumulativePercentages = getCumulativePercentages(data);
+		
+		assertEquals("Getting first value result", 0.4, actualCumulativePercentages.getValue(0));
+		assertEquals("Getting second value result", 0.4, actualCumulativePercentages.getValue(1));
```
Comment (enochtsang) [inline Tests/GetCumulativePercentagesTests.java]: These two should be `1/3` shouldn't they?

**Comment by enochtsang** [comment]
I'm not really understanding the expected values for the `GetCumulativePercentages`, can you explain?

**Comment by kkcmah** [comment]
From my interpretation of the javadoc provided the expected value is the sum of values from a given key to the first key divided by the sum of all input values. Let's say we have a input list like
Key Value
0     4
1     3
2     3
the sum of values would be 4 + 3 + 3 = 10
the expected value of key 0 is (key 0 value) / (sum of values) = (4) / (10) = 0.4
the expected value of key 1 is (key 1 value + key 0 value) / (sum of values) = (4 + 3) / (10) = 0.7 
the expected value of key 2 is (key 2 value + key 1 value + key 0 value) / (sum of values) = (4 + 3 + 3) / (10) = 1.0.

I assumed that null is 0. (8 + null) = 8.
In the cases where infinities were used as inputs the expected values were NaN because we're dividing by a infinity or end up with (infinity) / (infinity).

Oh yea... I just noticed that I should be checking the keys as well and possibly add a test case where the keys aren't in order.

**Comment by enochtsang** [comment]
Makes sense, looks good to me. @bryson189, @jqvillar ?

**Comment by bryson189** [comment]
Looks good.

**Pull request merged**